src/main/scala/com/actian/spark_vector/sql/VectorRelation.scala

```diff
@@ -122,4 +147,12 @@ object VectorRelation {
       ("", Nil)
     }
   }
+
+  /** Selects the subset of columns (represented by `ColumnMetadata` structures) as required by `requiredColumns` */
+  def pruneColumns(requiredColumns: Array[String], columnMetadata: Seq[ColumnMetadata]): Seq[ColumnMetadata] = if (requiredColumns.isEmpty) {
```

Comment:
nit: since this is out of this object's scope, maybe is a good idea to move it in the class and make it private?

i.e. requiredColumns comes through buildScan...don't see any use of it other that within that function :-/